services/app-web/src/components/ChangeHistory/ChangeHistoryV2.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react'
+import { dayjs } from '../../common-code/dateHelpers/dayjs'
+import { SectionHeader } from '../SectionHeader'
+import { Accordion, Link } from '@trussworks/react-uswds'
+import type { AccordionItemProps } from '@trussworks/react-uswds/lib/components/Accordion/Accordion'
+import { UpdateInformation, Contract } from '../../gen/gqlClient'
+import styles from './ChangeHistory.module.scss'
+type ChangeHistoryProps = {
+    contract: Contract
+}
+
+type flatRevisions = UpdateInformation & {
+    kind: 'submit' | 'unlock'
+    revisionVersion: string | undefined
+}
+
+export const ChangeHistoryV2 = ({
+    contract,
+}: ChangeHistoryProps): React.ReactElement => {
+    const flattenedRevisions = (): flatRevisions[] => {
+        const result: flatRevisions[] = []
+        //Reverse revisions to order from earliest to latest revision. This is to correctly set version for each
+        // contract & recontract.
+        const reversedRevisions = [...contract.packageSubmissions].reverse()
```

Comment:
🔍 Is this code working properly with linked rates?  

 I am under the impression that we would only want contract related package submissions in our change right now. I _think_ at this point we only seek parity with current `ChangeHistory` display (e.g. even if there are linked rates we are not yet  adding in linked rates to change history). To achieve this, we would need to only map contract changes from packageSubmissions array to this `reversedRevisions` array. 
 
 Change history work with the full package submissions list is entirely different epic I believe
 
 @macrael can you weigh in?

Reply:
I think that if you filtered out all packages that dont have reason: "CONTRACT_SUBMISSION" then you will have the same number of entries as before. That will also give us the correct unlock reasons, we should only ever show unlocks from CONTRACT_SUBMISSION, not from RATE* updates

I definitely want this code to be using packageSubmissions and get off of the old revisions code

Reply:
@haworku that makes sense to me

Reply:
@pearl-truss  sounds like @macrael agrees, we should be filtering only the contract_submission packages here - can you make that change and ignore rate stuff? Otherwise looks good.